osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/AdHocManagementServiceTest.java

```diff
@@ -0,0 +1,102 @@
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import org.mockito.stubbing.Answer;
+import org.opensmartgridplatform.adapter.domain.core.infra.jms.core.OsgpCoreRequestMessageSender;
+import org.opensmartgridplatform.domain.core.entities.Device;
+import org.opensmartgridplatform.domain.core.entities.Organisation;
+import org.opensmartgridplatform.domain.core.exceptions.UnknownEntityException;
+import org.opensmartgridplatform.domain.core.services.DeviceDomainService;
+import org.opensmartgridplatform.domain.core.services.OrganisationDomainService;
+import org.opensmartgridplatform.shared.exceptionhandling.ComponentType;
+import org.opensmartgridplatform.shared.exceptionhandling.FunctionalException;
+import org.opensmartgridplatform.shared.infra.jms.RequestMessage;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class AdHocManagementServiceTest {
+    @Mock
+    private DeviceDomainService deviceDomainService;
+    @Mock
+    private OsgpCoreRequestMessageSender osgpCoreRequestMessageSender;
+    @Mock
+    private Device device;
+    @Mock
+    private OrganisationDomainService organisationDomainService;
+    @InjectMocks
+    private AdHocManagementService adHocManagementService;
+
+    private static final int PRIORITY = 1;
+    private static final String MESSAGE_TYPE = "testType";
+    private static final String CORRELATION_UUID = "correlationUid";
+    private static final String DEVICE_IDENTIFICATION = "deviceIdentification";
+    private static final String ORGANISATION_IDENTIFICATION = "orgIdentification";
+    private static final String IP_ADDRESS = "127.0.0.1";
+
+    @BeforeEach
+    public void setup() throws Exception {
+        when(this.device.getIpAddress()).thenReturn("127.0.0.1");
+        when(this.deviceDomainService.searchActiveDevice(any(String.class), any(ComponentType.class))).thenReturn(this.device);
+    }
+    
+    @Test
+    public void testSetReboot() throws Exception {
+        this.givenOnlyTheIdentifiedOrganisationExists(ORGANISATION_IDENTIFICATION);
+
+        final ArgumentCaptor<RequestMessage> messageCaptor = ArgumentCaptor.forClass(RequestMessage.class);
+        this.adHocManagementService.setReboot(ORGANISATION_IDENTIFICATION, DEVICE_IDENTIFICATION, CORRELATION_UUID,
+                MESSAGE_TYPE,
+                PRIORITY);
+        verify(this.osgpCoreRequestMessageSender).send(messageCaptor.capture(), eq(MESSAGE_TYPE), eq(PRIORITY),
+                eq(IP_ADDRESS));
+
+        final RequestMessage message = messageCaptor.getValue();
+
+        assertThat(CORRELATION_UUID).isEqualTo(message.getCorrelationUid());
+        assertThat(ORGANISATION_IDENTIFICATION).isEqualTo(message.getOrganisationIdentification());
```

Comment:
Assertj assertions like these are of the form `assertThat(actual).isEqualTo(expected)`.
Here the actual and the expected value are swapped, which may lead to confusing error message if the test would fail.